packages/client/src/links/internals/httpUtils.ts

```diff
@@ -194,11 +196,14 @@ export function httpRequest(
         });
       })
       .catch((err) => {
+        done = true;
         reject(TRPCClientError.from(err, { meta }));
       });
   });
   const cancel = () => {
-    ac?.abort();
+    if (!done) {
```

Comment:
That works as a workaround, but I wonder why observable is even cancelling it in the first place? Or is it just a generic "cleanup" function?

Reply:
Yeah, there might be something to what you're saying here

Reply:
Two years since I wrote the observable-stuff, my memory is rust, but I think it's a generic cleanup, but could prob be improved